README.md

```diff
@@ -1,65 +1,112 @@
+in a Kubernetes cluster using the bpfd operator and CRDs.
 
 ## Why bpfd?
 
-bpfd seeks to solve the following problems:
-
-- To allow multiple XDP programs to share the same interface
-- To give administrators control over who can load programs and to allow them to
-  define rules for ordering of networking BPF programs
-- To allow programs to be loaded automatically at system launch time
-- To simplify the packaging and loading of BPF-based infrastructure software (i.e Kubernetes CNI plugins)
-
-## How does it work?
-
-bpfd is built using [Aya](https://aya-rs.dev) an BPF library written in Rust.
-It offers two ways of interaction:
-
-- `bpfctl`: a command line tool
-- using GRPC
-
-It is expected that humans will use `bpfctl` whereas other applications on the system wishing to load programs using
-bpfd will use the GRPC. This allows for API bindings to be generated in any language supported by protocol buffers.
-We are initially targeting Go and Rust.
-See [Tutorial](docs/tutorial.md) for some examples of starting `bpfd`, managing logs, and using `bpfctl`.
-
-In order to allow the attachment of multiple XDP programs to the same interface, we have implemented the
-[libxdp multiprog protocol](https://github.com/xdp-project/xdp-tools/blob/master/lib/libxdp/protocol.org).
-Offering this in bpfd allows for XDP applications whose loader was not using libxdp to benefit from this.
-We are also hoping to find a way for applications linked with libxdp to use bpfd instead if it's
-in use in the system.
-
-## Usage
-
-Load a sample XDP Program:
-
-```console
-cargo build
-sudo ./target/debug/bpfd&
-sudo ./target/debug/bpfctl load --path /path/to/xdp/program --section-name "pass" xdp -i wlp2s0 --priority 50
-```
+### Why BPF?
+
+BPF is a powerful general-purpose framework that allows running sandboxed
+programs in the kernel.  It can be used for many purposes, including networking,
+monitoring, tracing and security.
+
+### Why BPF in Kubernetes?
+
+Demand is increasing from both Kubernetes developers and users. Examples of bpf
+in kubernetes include:
+
+- [Cilium](https://cilium.io/) and
+  [Calico](https://www.tigera.io/project-calico/) CNIs 
+- [Pixe](https://px.dev/): Open source observability
+- [KubeArmor](https://kubearmor.io/): Container-aware Runtime Security
+  Enforcement System
+- [Blixt](https://github.com/Kong/blixt): Gateway Api L4 conformance
+  implementation
+- [NetObserv](https://github.com/netobserv): Open Source Operator for network
+  observability 
+
+### Challenges for BPF in Kubernetes
+
+- Requires privileged pods
+  - BPF-enabled apps require at least CAP_BPF permissions, and more (depending
+    on what they need to do).
+  - These permissions give the pods great powers.
+- Handling multiple BPF programs
+  - Not all libraries support multiple XDP programs sharing the same interface.
+  - When multiple BPF programs are used, they can interfere with each other in
+    unpredictable ways.
```

Comment:
I agree with your addition, but think the existing bullet is valid too.

Reply:
Maybe? I think it needs some additional qualification because as it stands it could mean that an XDP program could interfere with a Tracepoint program in unexpected ways.

Reply:
I could qualify it as:
- they can interfere with each other in unpredictable ways
+ some BPF programs can interfere with others in unpredictable ways.

However, for the example given, if I have a tracepoint in some code that I'm expecting to be hit based on receiving a packet, and if someone else installs an XDP program that drops those packets, then I might wonder why my tracepoint isn't getting hit.  In that case, it might be good to know that some other app installed that XDP program and I might want to investigate it.